be/src/connector_sink/hive_chunk_sink.cpp

```diff
@@ -97,6 +97,9 @@ StatusOr<std::unique_ptr<ConnectorChunkSink>> HiveChunkSinkProvider::create_chun
 
     std::unique_ptr<formats::FileWriterFactory> file_writer_factory;
     if (boost::iequals(ctx->format, formats::PARQUET)) {
+        // ensure hive compatibility since hive 3 and lower version accepts specific encoding
+        ctx->options[formats::ParquetWriterOptions::USE_LEGACY_DECIMAL_ENCODING] = "true";
+        ctx->options[formats::ParquetWriterOptions::USE_INT96_TIMESTAMP_ENCODING] = "true";
```

Comment:
why pass options as map instead of struct? it's hard to tell how many options are there from a map.